functional_tests/experiments/discovery.R

```diff
@@ -0,0 +1,217 @@
+set.seed(2022)
+library(foreach)
+library(doParallel)
+
+# Set Experiment Parameter
+n_rules <- 2
+sample_size <- 2000
+effect_sizes <- seq(0, 10, 0.2)
+confoundings <- c("nc","lc","nlc")
+ITE_estimators <- c("ipw","aipw","sipw","cf","bcf")
+n_seeds <- 48
+ratio_dis <- 0.5
+
+# Set Ground Truth
+{
+cdr <- c("x1>0.5 & x2<=0.5", "x5>0.5 & x6<=0.5",
+         "x4<=0", "x5<=0.5 & x7>0.5 & x8<=0.5")
+em <- c("x1","x2","x5","x6","x4","x7","x8")
+if (n_rules==2) {
+  cdr<-cdr[1:2]
+  em <- em[1:4]
+} else if (n_rules==4) {
+} else {stop(paste("Synthtic dataset with", n_rules,"rules has not been
+                    implemented yet. Set 'n_rules' equal to 2 or 4 (rules)."))}
+}
+
+# Set Method and Hyper Parameters
+{
+method_params <- list(ratio_dis = ratio_dis,
+                      ite_method_dis="aipw",
+                      ps_method_dis = "SL.xgboost",
+                      oreg_method_dis = "SL.xgboost",
+                      include_ps_dis = TRUE,
+                      ite_method_inf = "aipw",
+                      ps_method_inf = "SL.xgboost",
+                      oreg_method_inf = "SL.xgboost",
+                      include_ps_inf = TRUE,
+                      include_offset = FALSE,
+                      cate_method = "linreg",
+                      cate_SL_library = "SL.xgboost",
+                      filter_cate = TRUE,
+                      offset_name = NA,
+                      random_state = 3591)
+
+hyper_params <- list(intervention_vars = c(),
+                     ntrees_rf = 100,
+                     ntrees_gbm = 50,
+                     node_size = 20,
+                     max_nodes = 5,
+                     max_depth = 15,
+                     max_decay = 0,
+                     type_decay = 2,
+                     t_ext = 0.025,
+                     t_corr = 1,
+                     t_pvalue = 0.01,
+                     replace = TRUE,
+                     stability_selection = TRUE,
+                     cutoff = 0.9,
+                     pfer = 0.5,
+                     penalty_rl = 1)
+}
+
+# Set Cluster
+cl <- makeCluster(detectCores())
```

Comment:
This is not part of the official code. However, it is just a comment. A detected core is not necessarily the number of available cores to be used by the user. A better approach is providing it by the user.

Reply:
there is no automatic way to detect the number of cores?